README.md

```diff
@@ -172,15 +309,67 @@ pipe_config = {
     }
 }
 
-pipe = pipeline.create(pipe_type="naive", **pipe_config)
+pipe = pipeline.create(pipe_type="beam", **pipe_config)
 pipe.run()
 ```
 
-> **Warning**  
+All configured _main inputs_ are merged before processing,
+and the _outputs_ are written in each sink configured.
+In this pipeline, _side inputs_ are existing **open gaps** that can be closed while processing.
+
+You can see more configuration examples [here](config/). 
+
+
+> [!NOTE]
+> The key "options" can be used for custom configuration of each pipeline type.
+  For example, you can pass any option available in the [Apache Beam Pipeline Options]. 
+
+> [!CAUTION]
 > Date ranges are inclusive for the start date and exclusive for the end date.
 
 
-### Using from CLI:
+#### BigQuery output schema
+
+The schema for the output **gap events** table in BigQuery
+is defined in [pipe_gaps/pipeline/schemas/ais-gaps.json](/pipe_gaps/pipeline/schemas/ais-gaps.json).
+
+#### BigQuery data persistence pattern
+
+<div align="justify">
+
+When an **open gap** is closed,
+a new **gap** event is created.
+
+In the case of BigQuery output,
+this means that we are using a persistence pattern
+that matches the [slowly changing dimension] type 2 
+(always add new rows).
+In consequence, the output table can contain two gap events with the same `gap_id`:
+the old **open gap** and the current **closed _active_ gap**.
+The versioning of gaps is done with a timestamp `gap_version` field with second precision.
+
+To query all _active_ gaps,
+you will just need to query the last versions for every `gap_id`.
+For example:
+```sql
+SELECT *
```

Comment:
I'd also provide this shorter alternative:

```
SELECT *
FROM `world-fishing-827.scratch_tomas_ttl30d.pipe_ais_gaps_filter_no_overlapping_and_short`
QUALIFY ROW_NUMBER() OVER (PARTITION BY gap_id ORDER BY gap_version DESC) = 1
```

Reply:
Will do.